hack/make-rules/test-cmd-util.sh

```diff
@@ -3041,6 +3044,62 @@ run_daemonset_history_tests() {
   set +o errexit
 }
 
+run_statefulset_history_tests() {
+  set -o nounset
+  set -o errexit
+
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl(v1:statefulsets, v1:controllerrevisions)"
+
+  # Pre-condition: no statefulset exists
+  kube::test::get_object_assert statefulset "{{range.items}}{{$id_field}}:{{end}}" ''
+  
+  # Command: create a StatefulSet (revision 1)
+  kubectl apply -f hack/testdata/rollingupdate-statefulset.yaml --record "${kube_flags[@]}"
```

Comment:
I tried `--record` locally and it did work. We need to verify controllerrevisions' annotation here first, otherwise this test might be flaky (when undo is called before StS controller creates controllerrevisions)